File: YYReader/Services/HTMLChallengeDetector.swift
enum HTMLChallengeDetector {
    static func isChallenge(_ html: String) -> Bool {
        let sample = html.lowercased()
        return sample.contains("cf-chl-")
            || sample.contains("cloudflare challenge")
            || sample.contains("<title>just a moment")
            || sample.contains("enable javascript and cookies to continue")
    }
}

File: windows/YYReader.Windows.Core/Services/HtmlChallengeDetector.cs
namespace YYReader.Windows.Core.Services;

public static class HtmlChallengeDetector
{
    public static bool IsChallenge(string html)
    {
        var sample = html.ToLowerInvariant();
        return sample.Contains("cf-chl-", StringComparison.Ordinal)
            || sample.Contains("cloudflare challenge", StringComparison.Ordinal)
            || sample.Contains("<title>just a moment", StringComparison.Ordinal)
            || sample.Contains("enable javascript and cookies to continue", StringComparison.Ordinal);
    }
}
